lib/pacemaker/pcmk_cluster_queries.c

```diff
@@ -228,6 +230,47 @@ designated_controller_event_cb(pcmk_ipc_api_t *controld_api,
     data->rc = pcmk_rc_ok;
 }
 
+static void
+node_name_event_cb(pcmk_ipc_api_t *controld_api, enum pcmk_ipc_event event_type,
+                   crm_exit_t status, void *event_data, void *user_data)
+{
+    data_t *data = (data_t *) user_data;
+    pcmk__output_t *out = data->out;
+    const pcmk_controld_api_reply_t *reply = NULL;
+
+    int rc = validate_controld_reply(data, controld_api, event_type, status,
+                                     event_data, pcmk_controld_reply_info);
+
+    if (rc != pcmk_rc_ok) {
+        return;
+    }
+
+    reply = (const pcmk_controld_api_reply_t *) event_data;
+
+    if (reply->data.node_info.uname == NULL) {
+        out->err(out, "Node is not known to cluster");
+        data->rc = pcmk_rc_node_unknown;
+        return;
+    }
+
+    if (data->show_output) {
+        /* Using data->node_id outputs "local node" for some formats if
+         * pcmk__node_name() is called with node_id=0. Using the reply's node ID
+         * would show the true cluster node ID. Neither option is clearly better
+         * than the other.
+         */
```

Comment:
Basically, assuming that we requested nodeid 0 (local node) and that the true node ID is 1, text-like formats will show:
```
Node name: this-node (local node)
```
And XML will show:
```
<node-name id="0" name="this-node"/>
```

The alternatives would be:
```
Node name: this-node (nodeid=1)
<node-name id="1" name="this-node"/>
```

Reply:
I'm inclined to just drop show_output and the new message, since we don't have anything using it, and the caller can easily display it however they want

Reply:
That was actually what I did first. Then I thought "we're stuck with public API once we expose it, so might as well give the option."

Happy to drop it again.